### PR TITLE
added @example support for parameters / properties

### LINKED ISSUE #615: v3.x @example tag does not work on model properties
## Sorting

* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request

* I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior
@example tags placed on models should add examples to their respective properties

```ts
export interface TestInput {
    /**
     * a number prop
     * @example 2
     */
    number: number;
    string: string;
}
```

```json
"TestInput": {
  "properties": {
    "number": {
      "type": "number",
      "format": "double",
      "description": "a number prop",
      "example": 2
    },
    "string": {
      "type": "string"
    }
  },
  "required": [
    "number",
    "string"
  ],
  "type": "object",
  "additionalProperties": true
}
```

## Current Behavior
No example is generated for properties.

```ts
export interface TestInput {
    /**
     * a number prop
     * @example 2
     */
    number: number;
    string: string;
}
```
```json
"TestInput": {
  "properties": {
    "number": {
      "type": "number",
      "format": "double",
      "description": "a number prop"
    },
    "string": {
      "type": "string"
    }
  },
  "required": [
    "number",
    "string"
  ],
  "type": "object",
  "additionalProperties": true
}
```

## Context (Environment)
Version of the library: v3.x
Version of NodeJS: 12.16.1
* Confirm you were using yarn not npm: [x]

## Detailed Description
When writing complex models, it is much more ergonomic to include examples above individual properties.  Additionally, the inconsistency for @example support can be confusing.  This should be a fairly simple change.

## Breaking change?
Not that I can tell

### PR DESCRIPTION
**Pull request by jfrconley**
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
* [x] Have you written unit tests?
* [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
* [x] This PR is associated with an existing issue?

**Closing issues**
closes #615 

### If this is a new feature submission:

* [x] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Test plan**

Checks that @example tags on either properties or parameters generated examples.

### CODE REVIEW

File: src/metadataGeneration/tsoa.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export namespace Tsoa {
 
   export interface Parameter {
     parameterName: string;
+    example?: any;
```
Comment (WoH) [inline src/metadataGeneration/tsoa.ts]:
```suggestion
    example?: unknown;
```

File: src/metadataGeneration/tsoa.ts [hunk 2]
```diff
@@ -62,6 +63,7 @@ export namespace Tsoa {
     default?: any;
     description?: string;
     format?: string;
+    example?: any;
```
Comment (WoH) [inline src/metadataGeneration/tsoa.ts]:
```suggestion
    example?: unknown;
```

File: src/swagger/swagger.ts [hunk 3]
```diff
@@ -91,6 +91,7 @@ export namespace Swagger {
     in: 'query' | 'header' | 'path' | 'formData' | 'body';
     required?: boolean;
     description?: string;
+    example?: any;
```
Comment (WoH) [inline src/swagger/swagger.ts]:
```suggestion
    example?: unknown;
```

File: src/metadataGeneration/tsoa.ts (hunk 2 above)
Comment (jfrconley) [reply src/metadataGeneration/tsoa.ts]: I went ahead and move the type over. Any => unknown is a good conversion in general

**Pull request merged**